### PR TITLE
Refactor hip adapter to new logger

### PR DESCRIPTION
**Pull request by lplewa**

### CODE REVIEW

**Comment by lplewa** [comment]
I created an octopus merge b45470f9e1a5c4bacbc210d68e900858b82fae25 of all logs changes (#1359; #1347; #1345; #1342; #1032) and tested integration with [intel/llvm repository](https://github.com/intel/llvm/pull/12580/files)

File: source/adapters/hip/common.cpp [hunk 1]
```diff
@@ -8,7 +8,7 @@
 //
 //===----------------------------------------------------------------------===//
 #include "common.hpp"
-
+#include "logger/ur_logger.hpp"
```
Comment (hdelan) [inline source/adapters/hip/common.cpp:11]: Usually we keep a newline between `#include "..."`s and `#include <...>`s

**Review by hdelan — CHANGES_REQUESTED** [review]

**Comment by hdelan** [comment]
Could you rebase to re-run the CI? Some jobs failed last time round

File: source/adapters/hip/common.cpp [hunk 2]
```diff
@@ -99,19 +99,18 @@ void checkErrorUR(hipError_t Result, const char *Function, int Line,
...
-    ErrorName = hipGetErrorName(Result);
-    ErrorString = hipGetErrorString(Result);
-    std::cerr << "\nUR HIP ERROR:"
-              << "\n\tValue:           " << Result
-              << "\n\tName:            " << ErrorName
-              << "\n\tDescription:     " << ErrorString
-              << "\n\tFunction:        " << Function
-              << "\n\tSource Location: " << File << ":" << Line << "\n\n";
-  }
+  const char *ErrorString = nullptr;
```
Comment (hdelan) [inline source/adapters/hip/common.cpp:102]: Can we initialize this and `ErrorName` with the return vals of `hipGetErrorName` and `hipGetErrorString`?

File: source/adapters/hip/adapter.cpp [hunk 3]
```diff
@@ -10,14 +10,44 @@
...
+    std::cerr << msg << std::endl;
+  }
+
+  ~ur_legacy_sink() = default;
+};
+
+ur_adapter_handle_t_::ur_adapter_handle_t_()
+    : logger(logger::get_logger("hip")) {
+
+  if (std::getenv("UR_LOG_HIP") != NULL)
```
Comment (hdelan) [inline source/adapters/hip/adapter.cpp:42]:
```suggestion
  if (std::getenv("UR_LOG_HIP") != nullptr)
```

File: source/adapters/hip/adapter.cpp [hunk 4]
```diff
@@ -10,14 +10,44 @@
...
   std::atomic<uint32_t> RefCount = 0;
+  logger::Logger &logger;
+  ur_adapter_handle_t_();
 };
 
+class ur_legacy_sink : public logger::Sink {
+public:
+  ur_legacy_sink(std::string logger_name = "", bool skip_prefix = true)
+      : Sink(std::move(logger_name), skip_prefix) {
+    this->ostream = &std::cerr;
```
Comment (hdelan) [inline source/adapters/hip/adapter.cpp:28]: Taking the address of `std::cerr` seems like a strange thing to do. Why are we doing this?

**Review by hdelan — CHANGES_REQUESTED** [review]

File: source/adapters/hip/adapter.cpp (hunk 4 above)
Comment (lplewa) [reply source/adapters/hip/adapter.cpp:28]: This is part of abstraction - in logger we use abstract class 'sink', which can write to any kind of output stream. in this legacy sink we are writing to stderr, so we have to use cerr stream.

File: source/adapters/hip/common.cpp (hunk 2 above)
Comment (lplewa) [reply source/adapters/hip/common.cpp]: Done

File: source/adapters/hip/adapter.cpp (hunk 3 above)
Comment (lplewa) [reply source/adapters/hip/adapter.cpp]: done

**Review by hdelan — APPROVED** [review]
LGTM

**Pull request merged**